File: ERP-Simulator-Server-End/Message.py
# !/usr/bin/env python3
# -*- coding: utf-8 -*-
# create data : 2018/03/20
# author : Zhiquan.Wang
import logging
import json


def get_request_type(*, json_info):
    return json.loads(json_info)[JsonAttribute.msg_type]


class MessageList(object):
    message_list = ('register', 'login', 'logout', 'create_game')
    register = message_list[0]
    login = message_list[1]
    logout = message_list[2]


class JsonAttribute(object):
    msg_type = 'Type'
    msg_suc = 'IsSuccessful'
    extra_info = 'ExtraInfo'
    login_id = 'LoginID'
    user_name = 'UserName'
    user_pwd = 'UserPwd'
    initial_cash = 'InitialCash'
    management_fee = 'ManagementFee'
    name = 'Name'
    purchase_cost = 'PurchaseCost'
    capacity = 'Capacity'
    rent_cost = 'RentCost'
    ownership = 'Ownership'
    area = 'Area'
    construction_time = 'ConstructionTime'
    depreciation_rate = 'DepreciationRate'
    rent_to_purchase_cost = 'Rent2purchaseCost'
    develop_time = 'DevelopTime'
    develop_cost = 'DevelopCost'
    construction_cost = 'ConstructionCost'
    maintain_cost = 'MaintainCost'
    productivity = 'Productivity'
    transfer_time = 'TransferTime'
    transfer_cost = 'TransferCost'
    is_transfer = 'IsTransfer'
    price = 'Price'
    transport_time = 'TransportTime'
    emg_purchase_times = 'EmgPurchaseTimes'
    discount_rate = 'DiscountRate'
    team_id = "TeamID"
    team_name = "TeamName"
    leader_name = "LeaderName"
    member_list = "MemberList"
    game_name = "GameName"
    game_info = "GameInfo"
    game_rule = "GameRule"


class RequestInfo(object):
    def __init__(self, *, msg_type=None, extra_info=None, json_info=None):
        if json_info is None:
            self._msg_type = msg_type
            self._extra_info = extra_info
        else:
            tmp_dic = json.loads(json_info)
            self._msg_type = tmp_dic[JsonAttribute.msg_type]
            self._extra_info = tmp_dic[JsonAttribute.extra_info]

    @property
    def msg_type(self):
        return self._msg_type

    @msg_type.setter
    def msg_type(self, _type):
        self._msg_type = _type

    @property
    def extra_info(self):
        return self._extra_info

    @extra_info.setter
    def extra_info(self, _e_info):
        self._extra_info = _e_info


class ReplyInfo(object):
    def __init__(self, *, msg_suc, extra_info=''):
        self._msg_suc = msg_suc
        self._extra_info = extra_info

    @property
    def msg_suc(self):
        return self._msg_suc

    @msg_suc.setter
    def msg_suc(self, _suc):
        self._msg_suc = _suc

    @property
    def extra_info(self):
        return self._extra_info

    @extra_info.setter
    def extra_info(self, _e_info):
        self._extra_info = _e_info


class LoginReq(RequestInfo):
    def __init__(self, *, user_name='', user_pwd='', extra_info='', json_info=None):
        if json_info is None:
            RequestInfo.__init__(self, msg_type=MessageList.login, extra_info=extra_info)
            self.__user_name = user_name
            self.__user_pwd = user_pwd
        else:
            tmp_dic = json.loads(json_info)
            RequestInfo.__init__(self, msg_type=tmp_dic[JsonAttribute.msg_type],
                                 extra_info=tmp_dic[JsonAttribute.extra_info])
            self.__user_name = tmp_dic[JsonAttribute.user_name]
            self.__user_pwd = tmp_dic[JsonAttribute.user_pwd]

    @property
    def user_name(self):
        return self.__user_name

    @user_name.setter
    def user_name(self, _n):
        self.__user_name = _n

    @property
    def user_pwd(self):
        return self.__user_pwd

    @user_pwd.setter
    def user_pwd(self, _p):
        self.__user_pwd = _p

    def __dict__(self):
        return {JsonAttribute.msg_type: self._msg_type, JsonAttribute.user_name: self.__user_name,
                JsonAttribute.user_pwd: self.__user_pwd, JsonAttribute.extra_info: self._extra_info}

    def to_json(self):
        return json.dumps(self.__dict__())


class LoginRep(ReplyInfo):
    def __init__(self, msg_suc='', login_id='', extra_info='', json_info=None):
        if json_info is None:
            ReplyInfo.__init__(self, msg_suc=msg_suc, extra_info=extra_info)
            self.__login_id = login_id
        else:
            tmp_dic = json.loads(json_info)
            ReplyInfo.__init__(self, msg_suc=tmp_dic[JsonAttribute.msg_suc],
                               extra_info=tmp_dic[JsonAttribute.extra_info])
            self.__login_id = tmp_dic[JsonAttribute.login_id]

    @property
    def login_id(self):
        return self.__login_id

    @login_id.setter
    def login_id(self, _id):
        self.__login_id = _id

    def __dict__(self):
        return {JsonAttribute.msg_suc: self._msg_suc, JsonAttribute.login_id: self.__login_id,
                JsonAttribute.extra_info: self._extra_info}

    def to_json(self):
        return json.dumps(self.__dict__())


class LogoutReq(RequestInfo):
    def __init__(self, *, login_id='', extra_info='', json_info=None):
        if json_info is None:
            RequestInfo.__init__(self, msg_type=MessageList.logout, extra_info=extra_info)
            self.__login_id = login_id
        else:
            tmp_dic = json.loads(json_info)
            RequestInfo.__init__(self, msg_type=tmp_dic[JsonAttribute.msg_type],
                                 extra_info=tmp_dic[JsonAttribute.extra_info])
            self.__login_id = tmp_dic[JsonAttribute.login_id]

    @property
    def login_id(self):
        return self.__login_id

    @login_id.setter
    def login_id(self, _id):
        self.__login_id = _id

    def __dict__(self):
        return {JsonAttribute.msg_type: self._msg_type, JsonAttribute.login_id: self.__login_id,
                JsonAttribute.extra_info: self._extra_info}


class LogoutRep(ReplyInfo):
    def __init__(self, msg_suc, extra_info=''):
        ReplyInfo.__init__(self, msg_suc=msg_suc, extra_info=extra_info)

    def __dict__(self):
        return {JsonAttribute.msg_suc: self._msg_suc, JsonAttribute.extra_info: self._extra_info}

    def to_json(self):
        return json.dumps(self.__dict__())


class RegisterReq(RequestInfo):
    def __init__(self, *, msg_type='', user_name='', user_pwd='', extra_info='', json_info=None):
        if json_info is None:
            RequestInfo.__init__(self, msg_type=msg_type, extra_info=extra_info)
            self.__user_name = user_name
            self.__user_pwd = user_pwd
        else:
            tmp_dic = json.loads(json_info)
            RequestInfo.__init__(self, msg_type=tmp_dic[JsonAttribute.msg_type],
                                 extra_info=tmp_dic[JsonAttribute.extra_info])
            self.__user_name = tmp_dic[JsonAttribute.user_name]
            self.__user_pwd = tmp_dic[JsonAttribute.user_pwd]

    @property
    def user_name(self):
        return self.__user_name

    @user_name.setter
    def user_name(self, _n):
        self.__user_name = _n

    @property
    def user_pwd(self):
        return self.__user_pwd

    @user_pwd.setter
    def user_pwd(self, _p):
        self.__user_pwd = _p

    def __dict__(self):
        return {JsonAttribute.msg_type: self._msg_type, JsonAttribute.user_name: self.__user_name,
                JsonAttribute.user_pwd: self.__user_pwd, JsonAttribute.extra_info: self._extra_info}

    def to_json(self):
        return json.dumps(self.__dict__())


class RegisterRep(ReplyInfo):
    def __init__(self, msg_suc, extra_info=''):
        ReplyInfo.__init__(self, msg_suc=msg_suc, extra_info=extra_info)

    def __dict__(self):
        return {JsonAttribute.msg_suc: self._msg_suc, JsonAttribute.extra_info: self._extra_info}

    def to_json(self):
        return json.dumps(self.__dict__())


class CreateGameReq(RequestInfo):
    def __init__(self, msg_type='', extra_info='', *, game_name='', game_info=None, game_rule=None, json_info=None):
        if json_info is None:
            CreateGameReq.__init__(self, msg_type=msg_type, extra_info=extra_info)
            self.__game_name = game_name
            self.__game_info = game_info
            self.__game_rule = game_rule
        else:
            tmp_dic = json.loads(json_info)
            CreateGameReq.__init__(self, msg_type=tmp_dic[JsonAttribute.msg_type],
                                   extra_info=tmp_dic[JsonAttribute.extra_info])
            self.__game_name = tmp_dic[JsonAttribute.game_name]
            self.__game_info = tmp_dic[JsonAttribute.game_info]
            self.__game_rule = tmp_dic[JsonAttribute.game_rule]

    def __dict__(self):
        return {JsonAttribute.name: self._msg_type, JsonAttribute.game_name: self.__game_name,
                JsonAttribute.extra_info: self._extra_info}

    def to_json(self):
        return json.dumps(self.__dict__())

File: ERP-Simulator-Server-End/FontEnd/FontEnd/Message.cs
using System;
using Newtonsoft.Json;
namespace FontEnd {
    class MessageList {
        public const String Login = "login";
        public const String Logout = "logout";
        public const String Register = "register";
        public static String[] messageName = { Login, Logout, Register };
    }

    abstract class RequestInfo {
        protected String type;
        protected String extraInfo;
        public RequestInfo() {

        }
        public RequestInfo(String _type) {
            type = _type;
            extraInfo = "";
        }

        public string ExtraInfo { get => extraInfo; set => extraInfo = value; }
        public string Type { get => type; }
        public String JsonSerialization() {
            return JsonConvert.SerializeObject(this);
        }
    }

    abstract class ReplyInfo {
        private Boolean isSuccessful;
        private String extraInfo;
        public ReplyInfo() {
        }
        public ReplyInfo(Boolean _b, String _e) {
            isSuccessful = _b;
            extraInfo = _e;
        }
        public bool IsSuccessful { get => isSuccessful; set => isSuccessful = value; }
        public String ExtraInfo { get => extraInfo; set => extraInfo = value; }
        public String JsonSerialization() {
            return JsonConvert.SerializeObject(this);
        }
    }


    class LoginReq : RequestInfo {
        private String userName;
        private String userPwd;
        public LoginReq() : base() {
            type = MessageList.Login;
        }
        public LoginReq(String _jsonS) {
            LoginReq tmp_l = JsonConvert.DeserializeObject<LoginReq>(_jsonS);
            type = MessageList.Login;
            userName = tmp_l.UserName;
            userPwd = tmp_l.UserPwd;
        }
        public LoginReq(String _n, String _p) : base(MessageList.Login) {
            userName = _n;
            userPwd = _p;
        }
        public string UserName { get => userName; set => userName = value; }
        public string UserPwd { get => userPwd; set => userPwd = value; }

    }

    class LoginRep : ReplyInfo {
        private String loginID;
        public LoginRep() : base() {

        }
        public LoginRep(Boolean _b, String _e = "") : base(_b, _e) {
            LoginID = "";
        }
        public LoginRep(String _jsonS) {
            LoginRep tmp_l = JsonConvert.DeserializeObject<LoginRep>(_jsonS);
            IsSuccessful = tmp_l.IsSuccessful;
            LoginID = tmp_l.LoginID;
            ExtraInfo = tmp_l.ExtraInfo;
        }
        public String LoginID { get => loginID; set => loginID = value; }
    }


    class LogoutReq : RequestInfo {
        String loginID;
        public LogoutReq() {
            type = MessageList.Logout;
        }
        public LogoutReq(String _id) : base(MessageList.Logout) {
            loginID = _id;
        }
        public string LoginID { get => loginID; set => loginID = value; }
    }

    class LogoutRep : ReplyInfo {
        public LogoutRep() : base() {

        }
        public LogoutRep(Boolean _b, String _e) : base(_b, _e) {

        }
        public LogoutRep(String _jsonS) {
            LogoutRep tmp_l = JsonConvert.DeserializeObject<LogoutRep>(_jsonS);
            IsSuccessful = tmp_l.IsSuccessful;
            ExtraInfo = tmp_l.ExtraInfo;
        }
    }
    class RegisterReq : RequestInfo {
        String userName;
        String userPwd;
        public RegisterReq() {
            type = MessageList.Register;
        }
        public RegisterReq(String _n, String _p) : base(MessageList.Register) {
            UserName = _n;
            UserPwd = _p;
        }

        public string UserName { get => userName; set => userName = value; }
        public string UserPwd { get => userPwd; set => userPwd = value; }
    }

    class RegisterRep : ReplyInfo {
        public RegisterRep() : base() {

        }
        public RegisterRep(Boolean _b, String _e) : base(_b, _e) {

        }
        public RegisterRep(String _jsonS) {
            RegisterRep tmp_l = JsonConvert.DeserializeObject<RegisterRep>(_jsonS);
            IsSuccessful = tmp_l.IsSuccessful;
            ExtraInfo = tmp_l.ExtraInfo;
        }
    }
    class CreateGameReq : RequestInfo {
        String gameName;
        GameInfo gameInfo;
        GameRule gameRule;
        public CreateGameReq() : base() {

        }
        public CreateGameReq(String _game_name, GameInfo _gInfo, GameRule _gRule) {
            GameName = _game_name;
            GameInfo = _gInfo;
            GameRule = _gRule;
        }

        public GameInfo GameInfo { get => gameInfo; set => gameInfo = value; }
        public GameRule GameRule { get => gameRule; set => gameRule = value; }
        public string GameName { get => gameName; set => gameName = value; }
    }

    class CreateGameRep : ReplyInfo {
        String gameID;

        public CreateGameRep() : base() {

        }
        public CreateGameRep(bool _b, String _e) : base(_b, _e) {

        }
        public CreateGameRep(String _jsonS) {
            CreateGameRep tmp_p = JsonConvert.DeserializeObject<CreateGameRep>(_jsonS);
            IsSuccessful = tmp_p.IsSuccessful;
            GameID = tmp_p.GameID;
            ExtraInfo = tmp_p.ExtraInfo;
        }
        public string GameID { get => gameID; set => gameID = value; }
    }
    
   
   
 
}